### PR TITLE
UML-677: Changes to add message for LPAs with instructions and preferences on dashboard

### PR DESCRIPTION
**Pull request by SeemaMenon**
## Purpose
context_https://opgtransform.atlassian.net/browse/UML-677

Fixes UML-677

## Approach
Made content addition on dashboard for LPAs with instructions and preferences.
Added a new page for instructions and preferences, page handler and new feature tests to test the additions.

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Refunds service_

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [X] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-front/app/features/context/UI/AccountContext.php [hunk 1]
```diff
@@ -2737,4 +2739,85 @@ public function iCanSeeUserAccountsTable()
...
+            ->respondWith(
+                new Response(
+                    StatusCodeInterface::STATUS_OK,
+                    [],
+                    json_encode([])));
+
+        $this->ui->visit('/lpa/dashboard');
+
+        $this->ui->assertPageAddress('/lpa/dashboard');
+        $this->ui->assertPageContainsText('Important: This lpa has instructions or preferences');
```
Comment (williamfalconeruk) [inline service-front/app/features/context/UI/AccountContext.php:2770]: I think this needs to be tied up the `$message` parameter

File: service-front/app/src/Actor/src/Handler/InstructionsPreferencesHandler.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    /**
+     * Handles a request and produces a response
+     *
+     * @param ServerRequestInterface $request
+     * @return ResponseInterface
+     */
+    public function handle(ServerRequestInterface $request): ResponseInterface
+    {
+        $actorLpaToken = "null";
+        $array = $request->getQueryParams();
```
Comment (GemTay) [inline service-front/app/src/Actor/src/Handler/InstructionsPreferencesHandler.php:48]: If the query parameters are empty, an exception should be thrown, as the actorLpaToken is required

File: service-front/app/src/Actor/templates/actor/partials/lpa-details.html.twig [hunk 3]
```diff
@@ -17,6 +17,11 @@
                     <a class="govuk-link" href="{{ path('lpa.access-codes', {}, {'lpa': actorToken }) }}">Check access codes</a><br>
                     <a class="govuk-link" href="{{ path('lpa.view', {}, {'lpa': actorToken}) }}">View LPA summary</a>
                 </p>
+                <p>
+                {% if lpa.applicationHasRestrictions == true or lpa.applicationHasGuidance == true %}
```
Comment (GemTay) [inline service-front/app/src/Actor/templates/actor/partials/lpa-details.html.twig:21]: You can remove the '== true' to simplify this

File: service-front/app/features/context/UI/AccountContext.php (hunk 1 above)
Comment (SeemaMenon) [reply service-front/app/features/context/UI/AccountContext.php:2770]: done

File: service-front/app/src/Actor/src/Handler/InstructionsPreferencesHandler.php (hunk 2 above)
Comment (SeemaMenon) [reply service-front/app/src/Actor/src/Handler/InstructionsPreferencesHandler.php:48]: cleaned

File: service-front/app/features/context/UI/AccountContext.php [hunk 4]
```diff
@@ -2737,4 +2739,85 @@ public function iCanSeeUserAccountsTable()
...
+     * @When /^I am on the instructions and preferences page$/
+     */
+    public function iAmOnTheInstructionsAndPreferencesPage()
+    {
+        $this->iAmOnTheDashboardPage();
+        $this->iClickTheReadMoreLinkInTheInstructionsOrPreferenceMessage('Read more');
+        $this->iAmNavigatedToTheInstructionsAndPreferencesPage();
+
+
+
```
Comment (GemTay) [inline service-front/app/features/context/UI/AccountContext.php:2821]: Blank lines to be removed

**Comment by SeemaMenon** [comment]
> Couple of small changes

@GemTay  - review changes applied and ready for a re-review please.

**Pull request merged**